### PR TITLE
OSIS-137 add canonical ID as owner of buckets

### PR DESCRIPTION
**Pull request by anurag4DSB**
Use Canonical ID for ownership of buckets as VMware now supports true IAM with OSIS 2.2.0.1

### CODE REVIEW

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java [hunk 1]
```diff
@@ -831,32 +831,33 @@ public static String maskSecretKey(String logStatement) {
...
      * @return the OSIS bucket meta
      */
-    public static OsisBucketMeta toOsisBucketMeta(Bucket bucket, String userId) {
+    public static OsisBucketMeta toOsisBucketMeta(Bucket bucket, String canonicalId) {
 
         return new OsisBucketMeta()
                 .name(bucket.getName())
                 .creationDate(bucket.getCreationDate().toInstant().toString())
-                .userId(userId);
+                .userId(canonicalId);
```
Comment (RachedBenMustapha) [inline osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:842]: What's the backward compatibility implication, if what was a userId is now the canonical id? What was used as a user id before?

Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:842]: This was never implemented properly by VMware OSE side. Previously we were sending account ID as owner/user ID and they were expecting buckets to be owned by IAM users like in ceph model. Hence this API never worked with VCD and as a result end users were never able to synchronize buckets in VCD. We collaborated with VMware engg last year for 4+ months helping them get true IAM support for OSE.  Per IAM spec they now support buckets being owned by accounts ie canonical ID. Now this API will be honored in OSE2.2.0.1 for true IAM style ownership.

Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:842]: As this API was never used by OSE I do not see any backward compatibility support needed for this. OSE does rebuild database entries on trying bucket synchronization on OSE(tested by us and VMware engg). It maintains its own entries for ownership in a Postgres server.

Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:842]: This is the new property `iam: true` in getInformation API, which tells OSE (support from v2.2.0.1+) to flip the switch from ceph to supporting IAM standard.

**Review by XinLiScality — APPROVED** [review]

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java [hunk 2]
```diff
@@ -1151,21 +1151,23 @@ public PageOfOsisBucketMeta getBucketList(String tenantId, long offset, long lim
             final AmazonS3 s3Client = this.s3.getS3Client(tempCredentials,
                     appEnv.getRegionInfo().get(0));
 
+            // Get account info by tenant ID for canonical ID
+            GetAccountRequestDTO getAccountRequest = ScalityModelConverter.toGetAccountRequestWithID(tenantId);
+            logger.debug("[Vault]GetAccount request:{}", new Gson().toJson(getAccountRequest));
+
+            AccountData accountData = vaultAdmin.getAccount(getAccountRequest);
+            logger.debug("[Vault]GetAccount response:{}", new Gson().toJson(accountData));
```
Comment (williamlardier) [inline osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java:1159]: Nit:

```suggestion
            logger.debug("[Vault] GetAccount request:{}", new Gson().toJson(getAccountRequest));

            AccountData accountData = vaultAdmin.getAccount(getAccountRequest);
            logger.debug("[Vault] GetAccount response:{}", new Gson().toJson(accountData));
```

**Review by williamlardier — APPROVED** [review]
LGTM, thanks for the explanation on the backward compatibility changes

**Pull request merged**